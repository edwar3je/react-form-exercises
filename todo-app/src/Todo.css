.todo-container {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    margin-top: 10px;
}

.remove-button {
    width: 20px;
    height: 20px;
    color: black;
    background-color: rgb(204, 88, 88);
    border-radius: 10%;
    font-size: 10px;
    margin-left: 10px;
}