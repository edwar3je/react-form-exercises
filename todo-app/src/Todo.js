import './Todo.css';

const Todo = ({task, remove, id}) => {
    
    const removeTodo = () => {
        alert(`Task: "${task}" removed.`)
        remove(id);
    }

    const string = `${task[0]}${task[1]}${task[2]}`;
    
    return (
        <div className="todo-container">
            <p>{task}</p>
            <button className="remove-button" id={string} onClick={removeTodo}>X</button>
        </div>
    );
}

export default Todo;