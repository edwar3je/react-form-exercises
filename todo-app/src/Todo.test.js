import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';

it("renders without crashing", () => {
    render(
        <Todo 
            task="make the bed."
            remove={() => console.log("you just removed a task")}
            id="somestring"
        />
    );
});

it("matches the snapshot", () => {
    const { asFragment } = render(
        <Todo 
            task="make the bed."
            remove={() => console.log("you just removed a task")}
            id="somestring"
        />
    );
    expect(asFragment()).toMatchSnapshot();
});