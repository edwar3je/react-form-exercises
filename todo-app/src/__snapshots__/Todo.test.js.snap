// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="todo-container"
  >
    <p>
      make the bed.
    </p>
    <button
      class="remove-button"
      id="mak"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
